doc/cygnus-ngsi/integration/orion_cygnus_spark.md

```diff
@@ -0,0 +1,216 @@
+#<a name="top"></a>Connecting Orion Context Broker with Spark Streaming, using Cygnus
+Content:
+
+* [Introduction](#section1)
+* [Setting up Orion Context Broker](#section2)
+* [Setting up Cygnus](#section3)
+* [Setting up Spark Streaming](#section4)
+* [Testing everything together](#section5)
+* [Further information](#section6)
+
+##<a name="section1"></a>Introduction
+The idea behind this document is to explain, step-by-step, how to send on change context information in a real time basis from Orion Context Broker to Spark Streaming, for its analysis.
+
+Such a connection will be made thanks to Cygnus, specifically by using the following components:
+
+* Flume's native Http source, listening on TCP/5050 port. 
+* Custom NGSI rest handler from Cygnus.
+* Flume's native memory channel.
+* Flume's native Avro sink, sending Avro binary envents to TCP/5051 port.
+
+The final architecture looks like:
+
+![](./img/orion_cygnus_spark.png)
+
+All components installation, configuration and running will be explained according to this document purposes. For the sake of simplicity, all the components will be deployed in a single local machine (even apache Spark, specifically designed for running on top of a cluster, but allowing local mode as well).
+
+[Top](#top)
+
+##<a name="section2"></a>Setting up Orion Context Broker
+Orion Context Broker installation guide can be fund [here](https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/admin/install.md#installation).
+
+The default configuarion is OK for this integration example. Simply run it by typing in a shell:
+
+```
+$ sudo service contextBroker start
+Starting contextBroker...                                    [  OK  ]
+```
+
+As usual, a suscription must be made in order Orion Context Broker sends notifications regarding certain context entity updates to Cygnus (typical room1/room entity is used in the example):
+
+```
+$ curl -X POST "http://localhost:1026/v2/subscriptions" -s -S -H "Content-Type: application/json" -d @- <<EOF
+{
+  "subject": {
+    "entities": [
+      {
+        "id": "room1",
+        "type": "room"
+      }
+    ]
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:5050/notify"
+    },
+    "attrsFormat": "legacy"
+  }
+}
+EOF
+```
+
+Context entities are added to Orion using this other command (typical room1/room entity is used in the example):
+
+```
+curl -X POST "http://localhost:1026/v2/entities" -s -S -H 'Content-Type: application/json' -d @- <<EOF
+{
+  "id": "room1",
+  "type": "room",
+  "temperature": {
+    "value": 26.5,
+    "type": "Float"
+  }
+}
+EOF
+```
+
```

Comment:
The request to update is slightly different and I suggest to include it for the sake of completeness. In particular:

```
curl -X POST "http://localhost:1026/v2/entities/room1/attrs" -s -S -H 'Content-Type: application/json' -d @- <<EOF
{
  "temperature": {
    "value": 27.5,
    "type": "Float"
  }
}
EOF
```

(Actually, there are several ways of updating, but I'd suggest to use the above one)